syncd/syncd_flex_counter.h

```diff
@@ -206,11 +206,13 @@ class FlexCounter
         std::set<std::string> m_priorityGroupPlugins;
         std::set<std::string> m_bufferPoolPlugins;
 
-        std::atomic_bool m_runFlexCounterThread = { false };
-        std::shared_ptr<std::thread> m_flexCounterThread = nullptr;
+        bool m_runFlexCounterThread = false;
+        std::unique_ptr<std::thread> m_flexCounterThread = nullptr;
```

Comment:
why unique ? this is not needed

Reply:
From the FlexCounter object perspective, if it is destroyed, it will call endFlexCounterThread() to join() the thread object. If the thread object is referenced by multiple FlexCounter objects, join() can be called multiple times on each FlexCounter destruction. As a thread object can only be joined once, I think we should eliminate such a possibility. From the memory management perspective, it makes no difference using either shared_ptr or unique_ptr

Reply:
are there multiple threads per counters? problem with unique pointer is that it can be hard to track where actual pointer is held if something wrong will happen

Reply:
Changed to shared_ptr